kyaml/fn/framework/frameworktestutil/frameworktestutil.go

```diff
@@ -133,64 +132,190 @@ func (rc ResultsChecker) compare(t *testing.T, path string) {
 
 		err = cmd.Execute()
 
-		// Compae the results
-		if expectedError != "" && !assert.Error(t, err, actualOutput.String()) {
-			if !rc.UpdateExpectedFromActual {
-				t.FailNow()
+		// Update the fixtures if configured to
+		if rc.UpdateExpectedFromActual {
+			if len(actualError.String()) > 0 {
```

Comment:
supernitty nit: this could be `if actualError.String() == ""` which is more idiomatic and exactly as performant.